test/Microsoft.SqlTools.ManagedBatchParser.IntegrationTests/BatchParser/BatchParserTests.cs

```diff
@@ -232,15 +230,52 @@ public void VerifyLexerSetState()
                 }
                 executionResult = ScriptExecutionResult.Success;
             }
-            catch (Exception ex)
+            catch (Exception)
             {
-
                 executionResult = ScriptExecutionResult.Failure;
-
             }
             //  we doesn't expect any exception or testCase failures
             Assert.Equal<ScriptExecutionResult>(ScriptExecutionResult.Success, executionResult);
-             
+        }
+
+        // This test case is to verify that, Powershell's Invoke-SqlCmd handles ":!!if" in an inconsistent way
+        // Inconsistent way means, instead of throwing an exception as "Command Execute is not supported." it was throwing "Incorrect syntax near ':'."
```

Comment:
I would not try to explain what "inconsistent" mean: just don't use that word - 'cuz if you don't provide the context is  not easy to understand. Just say something like: "Verify that lexing/parsing !!if is consistent with the behavior of sqlcmd.exe, i.e. a space is not needed after an execute token ("!!").

This also shows that you are missing 1 test: they case where the is a space between "!!" and "if".